src/main/java/com/frameworkium/core/ui/pages/BasePage.java

```diff
@@ -100,15 +97,14 @@ public T get(long timeout) {
     @SuppressWarnings("unchecked")
     public T get() {
 
+        //Populate Page Object
         HtmlElementLoader.populatePageObject(this, driver);
 
-        // wait for page to load
-        if (isPageAngularJS()) {
-            waitForAngularRequestsToFinish();
-        }
+        //Wait for Elements & JS
         visibility.waitForAnnotatedElementVisibility(this);
+        javascriptWait.waitForJavascriptEventsOnLoad();
```

Comment:
It didn't detect Angular & Angular 2 the other way round. The only workaround I could think would be using a wait to poll the framework detection for a short fixed amount of time. Although, that's essentially hard-coding a wait for any page which doesn't use a JS framework.

With this, it'll wait for the elements to be present, which may mitigate the JS framework waits completely. Although, if that visible element is waiting to be populated after being visible, then it's still needed. 

It's unfortunate that detection doesn't work as soon as the body is ready.